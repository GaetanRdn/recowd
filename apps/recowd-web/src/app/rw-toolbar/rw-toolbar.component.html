<rc-toolbar>
  <img alt="avatar" rcAvatar src="assets/venom.png" />

  <rc-education-button-with-icon></rc-education-button-with-icon>

  <rc-experience-button-with-icon></rc-experience-button-with-icon>

  <rc-contact-button-icon></rc-contact-button-icon>
</rc-toolbar>
