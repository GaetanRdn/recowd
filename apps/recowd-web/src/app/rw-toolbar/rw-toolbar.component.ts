import { ChangeDetectionStrategy, Component } from '@angular/core';
import {
  AvatarDirective,
  ButtonDirective,
  MaterialIconDirective,
  ToolbarComponent,
} from '@recowd/ui-atoms';
import {
  ContactButtonIconComponent,
  EducationButtonWithIconComponent,
  ExperienceButtonWithIconComponent,
} from '@recowd/ui/molecules';

@Component({
  selector: 'rc-rw-toolbar',
  standalone: true,
  imports: [
    ToolbarComponent,
    AvatarDirective,
    ButtonDirective,
    MaterialIconDirective,
    ContactButtonIconComponent,
    EducationButtonWithIconComponent,
    ExperienceButtonWithIconComponent,
  ],
  templateUrl: './rw-toolbar.component.html',
  // styleUrls: ['./rw-toolbar-component.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class RwToolbarComponent {}
