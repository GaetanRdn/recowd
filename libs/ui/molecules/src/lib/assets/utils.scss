@function map-get-deep($map, $level, $subLevel) {
  @return map-get($map, $level, $subLevel);
}

@mixin use-typography($typography) {
  @each $key in font-size, font-family, font-weight, line-height {
    #{$key}: map-get($typography, $key);
  }
}
