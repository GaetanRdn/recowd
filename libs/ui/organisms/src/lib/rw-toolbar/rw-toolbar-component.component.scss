:host {
  rc-toolbar {
    gap: 2rem;

    rc-contact-button-icon {
      margin-left: auto;
    }
  }
}
