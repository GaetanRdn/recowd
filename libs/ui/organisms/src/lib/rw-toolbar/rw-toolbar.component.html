<rc-toolbar>
  <rc-rw-avatar></rc-rw-avatar>

  <rc-education-button-with-icon
    (click)="educationClick.next()"
  ></rc-education-button-with-icon>

  <rc-experience-button-with-icon
    (click)="experienceClick.next()"
  ></rc-experience-button-with-icon>

  <rc-contact-button-icon
    (click)="contactClick.next()"
  ></rc-contact-button-icon>
</rc-toolbar>
