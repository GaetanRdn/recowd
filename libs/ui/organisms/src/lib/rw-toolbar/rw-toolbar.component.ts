import {
  ChangeDetectionStrategy,
  Component,
  EventEmitter,
  Output,
} from '@angular/core';
import { ToolbarComponent } from '@recowd/ui-atoms';
import {
  ContactButtonIconComponent,
  EducationButtonWithIconComponent,
  ExperienceButtonWithIconComponent,
  RwAvatarComponent,
} from '@recowd/ui/molecules';

@Component({
  selector: 'rc-rw-toolbar',
  standalone: true,
  imports: [
    ToolbarComponent,
    RwAvatarComponent,
    ContactButtonIconComponent,
    EducationButtonWithIconComponent,
    ExperienceButtonWithIconComponent,
  ],
  templateUrl: './rw-toolbar.component.html',
  styleUrls: ['./rw-toolbar-component.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class RwToolbarComponent {
  @Output() public readonly educationClick: EventEmitter<void> =
    new EventEmitter<void>();
  @Output() public readonly experienceClick: EventEmitter<void> =
    new EventEmitter<void>();
  @Output() public readonly contactClick: EventEmitter<void> =
    new EventEmitter<void>();
}
