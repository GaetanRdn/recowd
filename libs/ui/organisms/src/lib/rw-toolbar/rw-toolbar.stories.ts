import { Meta, moduleMetadata, Story } from '@storybook/angular';
import { action } from '@storybook/addon-actions';
import { RwToolbarComponent } from './rw-toolbar.component';

export default {
  title: 'organisms/rw-toolbar',
  decorators: [
    moduleMetadata({
      imports: [RwToolbarComponent],
    }),
  ],
} as Meta<RwToolbarComponent>;

const Template: Story<RwToolbarComponent> = () => ({
  props: {
    experienceClick: action('experienceClick'),
    educationClick: action('educationClick'),
    contactClick: action('contactClick'),
  },
  template: `<rc-rw-toolbar (contactClick)="contactClick()" (educationClick)="educationClick()" (experienceClick)="experienceClick()"></rc-rw-toolbar>`,
});

export const Default = Template.bind({});
