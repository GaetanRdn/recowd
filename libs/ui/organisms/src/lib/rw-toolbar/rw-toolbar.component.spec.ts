import { Component } from '@angular/core';
import { TemplateLookup } from '@recowd/test/utils';
import { TestBed, waitForAsync } from '@angular/core/testing';
import { RwToolbarComponent } from './rw-toolbar.component';
import { TranslateModule } from '@ngx-translate/core';

describe('RwToolbarComponent', () => {
  let templateLookup: TemplateLookup<HostComponent>;

  beforeEach(waitForAsync(() => {
    TestBed.configureTestingModule({
      declarations: [HostComponent],
      imports: [RwToolbarComponent, TranslateModule.forRoot()],
    }).compileComponents();

    templateLookup = new TemplateLookup(HostComponent);
    templateLookup.detectChanges();
  }));

  it('should create', () =>
    expect(templateLookup.firstChildElement).toMatchSnapshot());

  it.each`
    method                 | selector
    ${'onContactClick'}    | ${'rc-contact-button-icon'}
    ${'onExperienceClick'} | ${'rc-experience-button-with-icon'}
    ${'onEducationClick'}  | ${'rc-education-button-with-icon'}
  `(
    'when click on $selector button then check HostComponent.$method is called',
    ({ method, selector }) => {
      // GIVEN
      jest.spyOn(templateLookup.hostComponent, method);

      // WHEN
      templateLookup.query(selector).click();

      // THEN
      expect(
        templateLookup.hostComponent[method as keyof HostComponent]
      ).toBeCalled();
    }
  );
});

@Component({
  template: ` <rc-rw-toolbar
    (contactClick)="onContactClick()"
    (educationClick)="onEducationClick()"
    (experienceClick)="onExperienceClick()"
  ></rc-rw-toolbar>`,
})
class HostComponent {
  public onExperienceClick(): void {
    // just spying
  }

  public onEducationClick(): void {
    // just spying
  }

  public onContactClick(): void {
    // just spying
  }
}
