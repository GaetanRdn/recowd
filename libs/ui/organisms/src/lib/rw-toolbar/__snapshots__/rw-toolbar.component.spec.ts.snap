// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RwToolbarComponent should create 1`] = `
<rc-rw-toolbar>
  <rc-toolbar>
    <rc-rw-avatars>
      <img
        alt="avatar"
        class="rc-avatar"
        rcavatar=""
        src="assets/venom.png"
      />
    </rc-rw-avatars>
    <rc-education-button-with-icon>
      <a
        class="rc-button"
        rcbutton=""
      >
        <rc-material-icon
          class="material-icons-outlined"
          type="outlined"
        >
          school
        </rc-material-icon>
         Formations

      </a>
    </rc-education-button-with-icon>
    <rc-experience-button-with-icon>
      <a
        class="rc-button"
        rcbutton=""
      >
        <rc-material-icon
          class="material-icons-outlined"
          type="outlined"
        >
          history_edu
        </rc-material-icon>
         Exp√©riences

      </a>
    </rc-experience-button-with-icon>
    <rc-contact-button-icon>
      <a
        class="rc-button"
        rcbutton=""
      >
        <rc-material-icon
          class="material-icons-outlined"
          type="outlined"
        >
          mail
        </rc-material-icon>
      </a>
    </rc-contact-button-icon>
  </rc-toolbar>
</rc-rw-toolbar>
`;
